packages/opentelemetry-plugin-postgres/src/utils.ts

```diff
@@ -0,0 +1,127 @@
+/*!
+ * Copyright 2019, OpenTelemetry Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { Span, CanonicalCode } from '@opentelemetry/types';
+import { AttributeNames } from './enums';
+import {
+  PgClientExtended,
+  PgPluginQueryConfig,
+  PostgresCallback,
+  PgClientConnectionParams,
+} from './types';
+import * as pgTypes from 'pg';
+import { PostgresPlugin } from './pg';
+
+export function _arrayStringifyHelper<T>(arr: Array<T>): string {
+  return '[' + arr.toString() + ']';
+}
+
+// Helper function to get a low cardinality command name from the full text query
+export function _getCommandFromText(text?: string): string {
+  if (text) {
+    const words = text.split(' ');
+    if (words && words.length > 0) {
+      return words[0];
+    }
+  }
+  return 'unknown';
+}
+
+export function _getJDBCString(params: PgClientConnectionParams) {
+  const host = params.host || 'localhost'; // postgres defaults to localhost
+  const port = params.port || 5432; // postgres defaults to port 5432
+  const database = params.database || '';
+  return `jdbc:postgresql://${host}:${port}/${database}`;
+}
+
+// Queries where args[0] is a QueryConfig
+export function _handleConfigQuery(
+  this: pgTypes.Client & PgClientExtended,
+  span: Span,
+  ...args: unknown[]
+) {
+  const argsConfig = args[0] as PgPluginQueryConfig;
+
+  // Set attributes
+  span.setAttribute(AttributeNames.DB_STATEMENT, argsConfig.text);
+  if (argsConfig.values instanceof Array) {
+    span.setAttribute(
+      AttributeNames.PG_VALUES,
+      _arrayStringifyHelper(argsConfig.values)
+    );
+  }
+  if (argsConfig.name) {
+    span.setAttribute(AttributeNames.PG_PLAN, argsConfig.name);
+  }
+
+  // Update span name with query command; prefer plan name, if available
+  const queryCommand = _getCommandFromText(argsConfig.name || argsConfig.text);
+  span.updateName(PostgresPlugin.BASE_SPAN_NAME + ':' + queryCommand);
```

Comment:
It is highly discouraged to update the name of a `Span` after its creation. 

Based on specs:
> The method name is called `UpdateName` to differentiate this method from the regular property setter. It emphasizes that this operation signifies a major change for a `Span` and may lead to re-calculation of sampling or filtering decisions made previously depending on the implementation. Alternatives for the name update may be late `Span` creation, when Span is started with the explicit timestamp from the past at the moment where the final Span name is known, or reporting a Span with the desired name as a child Span.

Is it possible to avoid this? Do we need to update the name with query command?

Reply:
It can be avoided. The spec calls for DB spans to be named with a low cardinality name describing the query, e.g. `insert` or `select`. I've moved the span creation to the query handlers so that I can add this information to the span on creation. bfb19d3

Reply:
Nice! Thanks for doing this 👍